src/Swarm/TUI/Model.hs

```diff
@@ -20,20 +20,33 @@ module Swarm.TUI.Model (
   Modal (..),
 
   -- * UI state
+
+  -- ** REPL
   REPLHistItem (..),
   firstReplEntry,
+  replItemText,
+  isREPLEntry,
+  getREPLEntry,
+  REPLHistory,
+  replSeq,
+  replStart,
+  replIndex,
```

Comment:
Having thought a little bit more about this I disagree with your assessment to some extent. I fully agree that if we accept that we need a `Getter` for `replIndex` it is not worth the trouble to just export a `Getter` as opposed to a full lens. 
However, I do think that we neither should need nor pragmatically do need to export even a `Getter` for `replIndex`. 

Pragmatically outside of `Model.hs` we only actually use `replIndex` in `adjReplHistIndex` in `Controller.hs` as well as in `drawREPL` in `View.hs`.  For the former function I propose a refactoring below that does not rely on `replIndex`. For the latter function we only need `replIndex` for a debugging functionality to print the index. So the lens could be replaced by a function exported from `Model.hs` returning a textual representation of the REPL index.

Conceptually I see no big difference between exporting `replIndex` and `replSeq` because both work in tandem. Without knowing about the internals of the storage of `REPLHistItem`s in `REPLHistory` like, e.g., the indexing scheme, the information contained in `replIndex` should be quite meaningless to a client. An example is its first use in `adjReplHistIndex` where it is compared agains the history length to determine if the input buffer is currently focused. I think there should rather be a dedicated function for this in `Model.hs` as part of the public contract around `REPLHistory`.